sympy/integrals/integrals.py

```diff
@@ -1328,6 +1335,11 @@ def as_sum(self, n=None, method="midpoint", evaluate=True):
         elif method == "trapezoid":
             result = dx*((f.subs(x, a) + f.subs(x, b))/2 +
                 Sum(f.subs(x, a + k*dx), (k, 1, n - 1)))
+        elif method == "simpson":
+            result = dx*((f.subs(x, a) + 
+                4 * Sum(f.subs(x, a + (2*k-1)*dx), (k, 1, n / 2)) +
+                2 * Sum(f.subs(x, a + 2*k*dx), (k, 1, (n / 2) - 1))) + 
```

Comment:
If `n` is odd then this will return invalid summations. It is probably better to treat `n` as meaning `2*n` steps.

Reply:
I added an exception when n is odd. I suggest that it is better to provide semantic consistency across all methods.